packages/vouching/test/Vouching.test.js

```diff
@@ -0,0 +1,265 @@
+          await assertRevert(
+            this.vouching.transferOwnership(dependencyName, ZERO_ADDRESS, { from: developer })
+          );
+        });
+
+        it('transfers the dependency\'s ownership to a given address', async function () {
+          const result = await this.vouching.transferOwnership(
+            dependencyName, transferee, { from: developer }
+          );
+
+          (await this.vouching.getDependencyOwner(dependencyName)).should.equal(transferee);
+          expectEvent.inLogs(
+            result.logs, 'OwnershipTransferred', { oldOwner: developer, newOwner: transferee }
+          );
+        });
+      });
+
+      describe('vouch', function () {
+        it('reverts when caller is not the dependency\'s owner', async function () {
+          await assertRevert(
+            this.vouching.vouch(dependencyName, stakeAmount, { from: vouchingOwner })
+          );
+        });
+
+        it('transfers stake amount of tokens from sender to vouching contract', async function () {
+          const vouchingInitBalance = await this.token.balanceOf(this.vouching.address);
+          const devInitBalance = await this.token.balanceOf(developer);
+
+          await this.vouching.vouch(dependencyName, stakeAmount, { from: developer });
+
+          (await this.token.balanceOf(this.vouching.address)).should.be.bignumber.equal(
+            vouchingInitBalance.plus(stakeAmount)
+          );
+          (await this.token.balanceOf(developer)).should.be.bignumber.equal(
+            devInitBalance.minus(stakeAmount)
+          );
+        });
+
+        it('adds the amount vouched to the existing dependency stake', async function () {
+          const initialStake = await this.vouching.getDependencyStake(dependencyName);
+
+          await this.vouching.vouch(dependencyName, stakeAmount, { from: developer });
+          await this.vouching.vouch(dependencyName, stakeAmount, { from: developer });
+
+          (await this.vouching.getDependencyStake(dependencyName)).should.be.bignumber.equal(
+            initialStake.plus(stakeAmount.times(2))
+          );
+        });
+
+        it('emits Vouched event', async function () {
+          const result = await this.vouching.vouch(dependencyName, stakeAmount, { from: developer });
+          const vouchedEvent = expectEvent.inLogs(result.logs, 'Vouched', {
+            name: dependencyName
+          });
+          vouchedEvent.args.amount.should.be.bignumber.equal(stakeAmount);
+        });
+      });
+
+      describe('unvouch', function () {
+        const safeUnstakeAmount = stakeAmount.minus(minStake);
+
+        it('reverts when caller is not the dependency\'s owner', async function () {
+          await assertRevert(
+            this.vouching.unvouch(dependencyName, safeUnstakeAmount, { from: vouchingOwner })
+          );
+        });
+
+        it('reverts when the remaining stake amount is less than the minimum', async function () {
```

Comment:
Can we add a test that checks attempting to unvouch more than the current stake, so we exercise the usage of safemath there?